src/app/components/Account/AccountLinkWithAddressSwitch.tsx

```diff
@@ -0,0 +1,39 @@
+import { FunctionComponent, useEffect, useState } from 'react'
+import { AddressSwitchOption } from '../AddressSwitch'
+import { AccountLink, AccountLinkProps } from './AccountLink'
+import { getOasisAddress, isValidEthAddress, isValidOasisAddress } from '../../utils/helpers'
+
+interface WithAddressSwitchProps {
+  addressSwitchOption: AddressSwitchOption
+}
+
+interface WrappedComponentProps extends AccountLinkProps, WithAddressSwitchProps {}
+
+const withAddressSwitch =
```

Comment:
what is a benefit of using HOC over a component here ?

Reply:
TODO for me: Make `withAddressSwitch` use a generic instead of `AccountLinkProps`, and move `AccountLinkWithAddressSwitch` to a separate file.